beetmoverscript/task.py

```diff
@@ -17,6 +19,27 @@ def validate_task_schema(context):
     scriptworker.client.validate_json_schema(context.task, task_schema)
 
 
+def validate_task_scopes(context, manifest):
+    # make sure scopes exist and are properly set
+    scopes = context.task['scopes']
+    for scope in scopes:
+        if scope.startswith("project:releng:beetmover:"):
+            signing_cert_name = scope.split(':')[-1]
+            if re.search('^[0-9A-Za-z_-]+$', signing_cert_name) is not None:
+                break
+            log.warning('scope {} is malformed, skipping!'.format(scope))
+    else:
+        log.critical("No beetmover scopes!")
+        sys.exit(3)
+
+    # prevent uncoordination between the bucket supposed to use for beetmove
+    # and the current credentials scopes
+    if (signing_cert_name not in manifest['s3_prefix_dated'] or
+            signing_cert_name not in manifest['s3_prefix_latest']):
```

Comment:
Thanks for adding this!

Now I'm worried about someone pushing to `pub/mobile/nightly/...` where `...` contains `dep` in the string, but isn't part of the path.

I'm not sure how hardcoded we want to go, but we could do something like

```python
restricted = {
    'pub/mobile/nightly': 'nightly',
    'pub/firefox/nightly': 'nightly',
    'pub/mobile/release': 'release',
    'pub/firefox/release': 'release',
}

for k, v in restricted:
    for path in (manifest['s3_prefix_dated'], manifest['s3_prefix_latest']):
        if path.startswith(k) and scope != v:
            # fatal error
```

Approved with that change :)
I'll go ahead and file a separate creds issue that we can do later.